Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -1508,18 +1207,16 @@ class CombinatorialKalmanFilter {
     std::vector<typename TrackContainer::TrackProxy> tracks;
 
     for (auto tip : combKalmanResult.lastMeasurementIndices) {
-      auto it = combKalmanResult.fittedParameters.find(tip);
-      if (it == combKalmanResult.fittedParameters.end()) {
-        continue;
-      }
-
       auto track = trackContainer.getTrack(trackContainer.addTrack());
       track.tipIndex() = tip;
 
-      const BoundTrackParameters& parameters = it->second;
-      track.parameters() = parameters.parameters();
-      track.covariance() = *parameters.covariance();
-      track.setReferenceSurface(parameters.referenceSurface().getSharedPtr());
+      if (auto it = combKalmanResult.fittedParameters.find(tip);
```

Comment:
1. So with this change we create tracks even if we don't have fitted parameters for them? 
2. Is this needed so we can do the smoothing? 
3. The fitted parameters are still at a reference surface, i.e. the perigee? We propagate there from the final surface, right?

I would imagine we still don't want to return tracks at all where the "fitted parameters" are empty, doesn't this mean that the target propagation failed?